hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -140,6 +146,48 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
     }
   }
 
+  private void trimArchiveFilesIfNecessary(HoodieEngineContext context) throws IOException {
+    Stream<HoodieLogFile> allLogFiles = FSUtils.getAllLogFiles(metaClient.getFs(),
+        archiveFilePath.getParent(),
+        archiveFilePath.getName(),
+        HoodieArchivedLogFile.ARCHIVE_EXTENSION,
+        "");
+    List<HoodieLogFile> sortedLogFilesList = allLogFiles.sorted(HoodieLogFile.getReverseLogFileComparator()).collect(Collectors.toList());
+    if (!sortedLogFilesList.isEmpty()) {
+      List<String> skipped = sortedLogFilesList.stream().skip(maxArchiveFilesToKeep).map(HoodieLogFile::getPath).map(Path::toString).collect(Collectors.toList());
+      if (!skipped.isEmpty()) {
+        LOG.info("Deleting archive files :  " + skipped);
+        context.setJobStatus(this.getClass().getSimpleName(), "Delete archive files");
+        Map<String, Boolean> result = deleteFilesParallelize(metaClient, skipped, context, true);
```

Comment:
remove local variable assignment since it's not used?

Reply:
Changed.